packages/xstate-wallet/src/workflows/application.ts

```diff
@@ -350,13 +331,39 @@ export const workflow = (
 
           // TODO: Catch other errors
           let message: ErrorResponse['error'];
-          if (matches(Errors.channelMissing)) message = {code: 403, message: 'Not your turn'};
-          else if (matches(Errors.notMyTurn)) message = {code: 400, message: 'Channel not found'};
+          if (matches(Errors.notMyTurn)) message = {code: 403, message: 'Not your turn'};
+          else if (matches(Errors.channelMissing))
+            message = {code: 400, message: 'Channel not found'};
           else {
             message = {code: 500, message: 'Wallet error'};
             console.error({error}, 'UpdateChannel call failed with error 500');
           }
 
+          messagingService.sendError(event.requestId, message);
+        }
+      }
+    },
+
+    closeChannel: async (context: ChannelIdExists, event: PlayerRequestConclude) => {
+      if (context.channelId === event.channelId) {
+        try {
+          messagingService.sendResponse(
+            event.requestId,
+            serializeChannelEntry(await store.updateChannel(event.channelId, {isFinal: true}))
+          );
+        } catch (error) {
+          const matches = reason => new RegExp(reason).test(error.message);
+
+          // TODO: Catch other errors
+          let message: ErrorResponse['error'];
+          if (matches(Errors.notMyTurn)) message = {code: 403, message: 'Not your turn'};
+          else if (matches(Errors.channelMissing))
+            message = {code: 400, message: 'Channel not found'};
+          else {
+            message = {code: 500, message: 'Wallet error'};
```

Comment:
So I arbitrarily chose to _not_ repeat error codes for different API calls. I don't know if that's the right choice, but we should either stick with it, and use the `CloseChannel` error codes, or change that choice.

https://github.com/statechannels/monorepo/blob/master/packages/client-api-schema/src/error-codes.ts#L8-L11